linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -2844,10 +2876,13 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
         // Atomic expressions
 
         case VarRef(name) =>
-          if (env.isLocalVar(name))
-            js.VarRef(transformLocalVarIdent(name))
-          else
+          if (env.isLocalVar(name)) {
+            env.localReplacement(name)
```

Comment:
Hmmm... Is it? I would expect the position to be the position of the original value, not of the capture (similar to when we inline a method, I'd copy the positions of the body and not re-position to the call site).

Reply:
But we're not inlining, here. We're using a "local" var, as a `VarRef`. That `VarRef`'s position is the position of where we use the value. And here it gets replaced by the position of the definition of the capture param, which is basically the position of the lambda.

In code, consider:
```scala
val x: Int = ???
val f = { (y: Int): Int => // <-- position of the lambda/capture param def
  x + y //  <-- position of the VarRef where we use `x`
}
```

Reply:
> by the position of the definition of the capture param

I guess you meant the position of the capture *value*, right? What would be the position of the capture *value* in your example? If it is the position of the lambda, then indeed the current approach would lead to worse positions.

The tricky bit is that under optimization, it can happen that a value is actually a literal. And in that case we'd lose the position of that literal :S (this is where I'm thinking about it more like inlining). There is an example in the [reversi gist](https://gist.github.com/gzm0/80f594aa454a93e0e83a758b5a66abc6#file-reversi-diff-L37) of this actually happening.

Reply:
Yes, the capture value and the capture definition always have the same position, which is that of the lambda. One captured variable can be used multiple times in the lambda, and it would only appear once in the capture defs/values, so this is unavoidable.

If it's a literal, it's unclear which position is best, actually.

Reply:
I'm not sure what to do about this.

Reply:
In practice, the result of `env.localReplacement(name)` will be either a `js.VarRef` or one of the `js.Literal`s, won't it? In that case, perhaps the least bad is to have a helper `def amendPos(tree: js.Tree, pos: Position): js.Tree` which pattern-matches `tree` against the likely ones, and returns a copy with the specified `pos`.